src/main/java/net/sf/jabref/gui/keyboard/KeyBindingsDialogViewModel.java

```diff
@@ -0,0 +1,196 @@
+
+    /**
+     * Read all keybindings from the keybinding repository and create table keybinding
+     * models for them
+     */
+    private void populateTable() {
+        TreeItem<KeyBindingViewModel> root = new TreeItem<>(new KeyBindingViewModel(KeyBindingCategory.FILE));
+        for (KeyBindingCategory category : KeyBindingCategory.values()) {
+            TreeItem<KeyBindingViewModel> categoryItem = new TreeItem<>(new KeyBindingViewModel(category));
+            keyBindingRepository.getKeyBindings().forEach((keyBinding, bind) -> {
+                if (keyBinding.getCategory() == category) {
+                    KeyBindingViewModel keyBindViewModel = new KeyBindingViewModel(keyBinding, bind,
+                            keyBinding.getCategory());
+                    TreeItem<KeyBindingViewModel> keyBindTreeItem = new TreeItem<>(keyBindViewModel);
+                    categoryItem.getChildren().add(keyBindTreeItem);
+                }
+            });
+            categoryItem.setExpanded(true);
+            root.getChildren().add(categoryItem);
+        }
+        root.setExpanded(true);
+        keyBindingsTable.setRoot(root);
+    }
+
+    private void bindColumnValues() {
+        actionColumn.setCellValueFactory(cellData -> cellData.getValue().getValue().nameProperty());
+        shortcutColumn.setCellValueFactory(cellData -> cellData.getValue().getValue().shownBindingProperty());
+        resetColumn.setCellFactory(
+                new Callback<TreeTableColumn<KeyBindingViewModel, String>, TreeTableCell<KeyBindingViewModel, String>>() {
+
+                    @Override
+                    public TreeTableCell<KeyBindingViewModel, String> call(
+                            TreeTableColumn<KeyBindingViewModel, String> p) {
+                        return new TreeTableCell<KeyBindingViewModel, String>() {
+
+                            @Override
+                            public void updateItem(String item, boolean empty) {
+                                super.updateItem(item, empty);
+                                KeyBindingViewModel viewModel = getTreeTableRow().getItem();
+                                if ((viewModel != null) && !viewModel.isCategory()) {
+                                    Text graphic = new Text(IconTheme.JabRefIcon.CLEANUP_ENTRIES.getCode());
+                                    graphic.getStyleClass().add("icon");
+                                    setGraphic(graphic);
+                                    setOnMouseClicked(evt -> {
+                                        resetKeyBinding(getTreeTableRow().getItem());
+                                    });
+
+                                }
+                                keyBindingsTable.refresh();
+                            }
+                        };
+                    }
+                });
+    }
+
+    private void resetKeyBinding(KeyBindingViewModel keyBindingViewModel) {
+        if (keyBindingViewModel.getKeyBinding() != null) {
```

Comment:
Move this code to KeyBindingViewModel, so that you simply can write keyBindingViewModel.resetToDefault();